modules/sor/utils/data.ts

```diff
@@ -0,0 +1,170 @@
+import { Chain, PrismaPoolType } from '@prisma/client';
+import { PrismaPoolAndHookWithDynamic, HookData } from '../../../prisma/prisma-types';
+import { prisma } from '../../../prisma/prisma-client';
+import { Cache } from 'memory-cache';
+import config from '../../../config';
+import { poolsToIgnore } from './constants';
+
+const cache = new Cache<
+    string,
+    { pools: PrismaPoolAndHookWithDynamic[]; underlyingTokens: { address: string; decimals: number }[] }
+>();
+const SOR_POOLS_CACHE_KEY = `sor:pools`;
+
+export async function getBasePoolsFromDb(
+    chain: Chain,
+    protocolVersion: number,
+    considerPoolsWithHooks: boolean,
+    poolIds?: string[],
+): Promise<{ pools: PrismaPoolAndHookWithDynamic[]; underlyingTokens: { address: string; decimals: number }[] }> {
+    const type = {
+        in: [
+            'WEIGHTED',
+            'META_STABLE',
+            'PHANTOM_STABLE',
+            'COMPOSABLE_STABLE',
+            'STABLE',
+            'FX',
+            'GYRO',
+            'GYRO3',
+            'GYROE',
+        ] as PrismaPoolType[],
+    };
+
+    if (poolIds?.length) {
+        return getPoolsByIds(chain, protocolVersion, type, poolIds);
+    }
+
+    const cacheKey = `${SOR_POOLS_CACHE_KEY}:${chain}:${protocolVersion}:${considerPoolsWithHooks}`;
+    const cached = cache.get(cacheKey);
+    if (cached) return cached;
+
+    const result = await getFilteredPools(chain, protocolVersion, considerPoolsWithHooks, type);
+
+    // cache for 10s
+    cache.put(cacheKey, result, 10 * 1000);
+    return result;
+}
+
+async function getPoolsByIds(chain: Chain, protocolVersion: number, type: { in: PrismaPoolType[] }, poolIds: string[]) {
+    const pools = await prisma.prismaPool.findMany({
+        where: { id: { in: poolIds }, chain, protocolVersion, type },
+        include: {
+            dynamicData: true,
+            tokens: { include: { token: true } },
+        },
+    });
+    const underlyingTokens = await getUnderlyingTokensFromDBPools(pools, chain);
+    return { pools, underlyingTokens };
+}
+
+async function getFilteredPools(
+    chain: Chain,
+    protocolVersion: number,
+    considerPoolsWithHooks: boolean,
+    type: { in: PrismaPoolType[] },
+) {
+    const poolIdsToExclude = config[chain].sor?.poolIdsToExclude ?? [];
+
+    const [pools, lbps] = await Promise.all([
+        getPrimaryPools(chain, protocolVersion, type, poolIdsToExclude),
+        getLiquidityBootstrappingPools(chain, protocolVersion, poolIdsToExclude),
+    ]);
```

Comment:
I was initially thinking, why not add the LIQUIDITY_BOOTSTRAPPING to the type in getBasePoolsFromDb. That way the pools would not have to get fetched separately. But if I understand it correctly this all happens, because depending on the considerPoolsWithHooks flag lbps would be thrown out as they are their own hook. Is this accurate?

Reply:
Its actually because we dont enforce the min TVL limit of $100 on LBPs

Reply:
Exactly. They don't follow the same rules that we use to filter pools. Since they are bootstrapping token launches, if we filter them by liquidity, they wouldn't be picked up.